File: samples/SSLGet.vb
' $Id: SSLGet.vb,v 1.1 2005/02/17 22:47:24 jeffreyphillips Exp $
' SSLGet.vb - demonstrate trivial SSL get capability
' compile with "vbc /r:../bin/LibCurlNet.dll /out:../bin/SSLGet.exe SSLGet.vb"
 
' usage: SSLGet url, e.g. SSLGet https://sourceforge.net

Imports System
Imports SeasideResearch.LibCurlNet

Module SSLGet

    Public Sub Main(ByVal args As String())
        Try
            Curl.GlobalInit(CURLinitFlag.CURL_GLOBAL_ALL)

            Dim easy As Easy
            easy = new Easy

            ' Set up write delegate
	    Dim wf As Easy.WriteFunction
            wf = new Easy.WriteFunction(AddressOf OnWriteData)

            ' and the SSL delegate
            Dim sf As Easy.SSLContextFunction
            sf = new Easy.SSLContextFunction(AddressOf OnSSLContext)

            ' and the rest of the cURL options
            easy.SetOpt(CURLoption.CURLOPT_URL, args(0))
            easy.SetOpt(CURLoption.CURLOPT_WRITEFUNCTION, wf)
            easy.SetOpt(CURLoption.CURLOPT_SSL_CTX_FUNCTION, sf)
            easy.SetOpt(CURLoption.CURLOPT_CAINFO, "ca-bundle.crt")

            easy.Perform()
            easy.Cleanup()

            Curl.GlobalCleanup()
        Catch ex As Exception
            Console.WriteLine(ex)
        End Try
    End Sub

    ' Called by libcURL.NET when it has data for your program
    Public Function OnWriteData(ByVal buf() As Byte, _
        ByVal size As Int32, ByVal nmemb As Int32, _
        ByVal extraData As Object) As Int32
        Console.Write(System.Text.Encoding.UTF8.GetString(buf))
        return size * nmemb
    End Function

    Public Function OnSSLContext(ByVal ctx As SSLContext, _
        ByVal extraData As Object) As CURLcode
        ' To do anything useful with the SSLContext object, you'll need
        ' to call the OpenSSL native methods on your own. So for this
        ' demo, we just return what cURL is expecting.
        return CURLcode.CURLE_OK
    End Function

End Module

File: samples/SSLGet.cs
// $Id: SSLGet.cs,v 1.1 2005/02/17 22:47:24 jeffreyphillips Exp $
// SSLGet.cs - demonstrate SSL get capability
// Compile with "csc /r:../bin/LibCurlNet.dll /out:../bin/SSLGet.exe SSLGet.cs"

// usage: SSLGet url, e.g. SSLGet https://sourceforge.net

using System;
using SeasideResearch.LibCurlNet;

class SSLGet
{
    public static void Main(String[] args)
    {
        try {
            Curl.GlobalInit((int)CURLinitFlag.CURL_GLOBAL_ALL);

            Easy easy = new Easy();

            Easy.WriteFunction wf = new Easy.WriteFunction(OnWriteData);
            easy.SetOpt(CURLoption.CURLOPT_WRITEFUNCTION, wf);

            Easy.SSLContextFunction sf = new Easy.SSLContextFunction(OnSSLContext);
            easy.SetOpt(CURLoption.CURLOPT_SSL_CTX_FUNCTION, sf);

            easy.SetOpt(CURLoption.CURLOPT_URL, args[0]);
            easy.SetOpt(CURLoption.CURLOPT_CAINFO, "ca-bundle.crt");

            easy.Perform();
            easy.Cleanup();

            Curl.GlobalCleanup();
        }
        catch(Exception ex) {
            Console.WriteLine(ex);
        }
    }

    public static Int32 OnWriteData(Byte[] buf, Int32 size, Int32 nmemb,
        Object extraData)
    {
        Console.Write(System.Text.Encoding.UTF8.GetString(buf));
        return size * nmemb;
    }

    public static CURLcode OnSSLContext(SSLContext ctx, Object extraData)
    {
        // To do anything useful with the SSLContext object, you'll need
        // to call the OpenSSL native methods on your own. So for this
        // demo, we just return what cURL is expecting.
        return CURLcode.CURLE_OK;
    }
}
